ciao-controller/api/api.go

```diff
@@ -585,6 +589,54 @@ func updateQuotas(c *Context, w http.ResponseWriter, r *http.Request) (Response,
 	return Response{http.StatusCreated, resp}, nil
 }
 
+// updateConfig implements the response for a configuration update
+// API call, checks for valid input and test new configuration value
+// previous to perform the change.
+func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) (Response, error) {
+	cfgResponse := types.ConfigUpdateResponse{Response: ""}
+	token := r.Header["X-Auth-Token"]
+	if len(token) == 0 {
+		cfgResponse.Response = "No token received in request"
+		return Response{http.StatusBadRequest, cfgResponse}, errors.New(cfgResponse.Response)
+	}
+	// get context to validate tenantID later
+	ctx := r.Context()
+	var cfgReq types.ConfigRequest
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		cfgResponse.Response = fmt.Sprintf("Cannot read request: %s", err.Error())
+		return Response{http.StatusInternalServerError, cfgResponse}, errors.New(cfgResponse.Response)
+	}
+	err = json.Unmarshal(body, &cfgReq)
+	if err != nil {
+		cfgResponse.Response = fmt.Sprintf("Cannot decode request: %s", err.Error())
+		return Response{http.StatusInternalServerError, cfgResponse}, errors.New(cfgResponse.Response)
+	}
+	err = c.Service.UpdateClusterConfig(ctx, cfgReq, token[0])
```

Comment:
It was originally written to use tenantID string instead of ctx, but for sake of unit testing `updateClusteConfig` it was later changed to use ctx, will revert this, but coverage will be harder to accomplish for this functions.